docs/lang/articles/basic/syntax.md

```diff
@@ -4,280 +4,484 @@ sidebar_position: 1
 
-Scope inside Taichi kernels and Taichi functions is called Taichi scope, and scope outside them is called Python scope.
+To differentiate the code for Taichi from the code for Python, we use the two decorators `@ti.kernel` and `@ti.func`:
 
-A Taichi kernel is the entrypoint of a Taichi program, and it is similar to a `__global__` function in CUDA. It can only be called inside Python scope.
+- Functions decorated with `@ti.kernel` are called kernels.
+- Functions decorated with `@ti.func` are called Taichi functions. 
 
-A Taichi function can only be called inside Taichi scope, and it is similar to a `__device__` function in CUDA.
+## Taichi scope vs. Python scope
 
-Major differences between Taichi kernels and Taichi functions are listed in the table below.
+We introduce the two terms "Taichi scope" and "Python scope" to make it easy to define the call range of a function or specify the effective range of a variable. So, before you proceed, familiarize yourself with these two terms. 
 
-| | Taichi kernels | Taichi functions |
-| :--- | :--- | :--- |
-| Can be called in | Python scope | Taichi scope |
-| Argument type annotation | Mandatory | Recommended |
-| Return type annotation | Mandatory| Recommended |
-| Return value | Scalar/Vector/Matrix | Arbitrary |
-| Max number of total elements in arguments | 8 (for OpenGL and CC) or 64 (other) | Unlimited |
-| Max number of return values in a return statement | 1 | Unlimited |
-| Max number of total elements in return values | 30 | Unlimited |
+### Taichi scope
 
+The code inside a kernel or a Taichi function is in the Taichi scope. The code in the Taichi scope is compiled by Taichi's runtime and executed in parallel on CPU or GPU devices for high-performance computation. 
 
+> Taichi scope corresponds to the *device side* in CUDA.  
 
+### Python scope
 
-## Taichi-scope vs Python-scope
+Code outside of the Taichi scope is in the Python scope. The code in the Python scope is native Python and executed by Python's virtual machine, *not* by Taichi's runtime.  
 
-Code decorated by `@ti.kernel` or `@ti.func` is in the **Taichi-scope**.
+> Python scope corresponds to the *host side* in CUDA. 
 
-They will be compiled by the Taichi compiler and executed on CPU or GPU devices in parallel with high performance.
+## Kernel
 
-:::note
-For people from CUDA, Taichi-scope = **device** side.
-:::
-
-Code outside `@ti.kernel` or `@ti.func` is in the **Python-scope**.
+A kernel is the entry point from which Taichi's runtime takes control and the smallest unit for runtime execution. You can define multiple kernels in your program, and each kernel is an *independent* child program. You call a kernel the same way you call a Python function, and you are allowed to switch back and forth between Taichi's runtime and Python's virtual machine.
```

Comment:
I'm a bit confused by "child program"

Reply:
This is not an exact word, we just wanted to convey the idea that kernels are like "unit sub-programs" here.